crates/ruma-macros/src/events/event_enum.rs

```diff
@@ -474,7 +517,70 @@ fn expand_accessor_methods(
                     }),
                 }
             }
+        };
+
+        if kind == EventKind::State {
+            let full_content_enum = kind.to_full_content_enum();
+            let full_content_variants: Vec<_> =
+                variants.iter().map(|v| v.ctor(&full_content_enum)).collect();
+
+            accessors = quote! {
+                #accessors
+
+                /// Returns the content of this state event.
+                pub fn content(&self) -> #full_content_enum {
+                    match self {
+                        #(
+                            #self_variants(event) => match event {
+                                #ruma_common::events::#event_struct::Original(ev) => #full_content_variants(
+                                    #ruma_common::events::FullStateEventContent::Original {
+                                        content: ev.content.clone(),
+                                        prev_content: ev.unsigned.prev_content.clone()
+                                    }
+                                ),
+                                #ruma_common::events::#event_struct::Redacted(ev) => #full_content_variants(
+                                    #ruma_common::events::FullStateEventContent::Redacted(
+                                        ev.content.clone()
+                                    )
+                                ),
+                            }
+                        )*
+                        Self::_Custom(event) => match event {
+                            #ruma_common::events::#event_struct::Original(ev) => {
+                                #full_content_enum::_Custom {
+                                    event_type: crate::PrivOwnedStr(
+                                        ::std::convert::From::from(
+                                            ::std::string::ToString::to_string(
+                                                &#ruma_common::events::EventContent::event_type(
+                                                    &ev.content,
+                                                ),
+                                            ),
+                                        ),
+                                    ),
+                                    redacted: false,
+                                }
+                            }
+                            #ruma_common::events::#event_struct::Redacted(ev) => {
+                                #full_content_enum::_Custom {
+                                    event_type: crate::PrivOwnedStr(
+                                        ::std::convert::From::from(
```

Comment:
I guess this is copied from elsewhere? `.into_boxed_str()` might be more readable.